canal/canal_test.go

```diff
@@ -171,3 +171,23 @@ func TestRenameTableExp(t *testing.T) {
 		}
 	}
 }
+
+func TestDropTableExp(t *testing.T) {
+	cases := []string{
+		"drop table test1",
+		"DROP TABLE test1",
+		"DROP table test.test1 IF EXISTS test.test1",
+		"drop table `test1`",
+		"DROP TABLE `test1`",
+		"DROP table `test`.`test1` IF EXISTS `test`.`test1`",
+		"DROP TABLE `test1` /* generated by server */",
+	}
+
+	table := []byte("test1")
+	for _, s := range cases {
+		m := expDropTable.FindSubmatch([]byte(s))
+		if m == nil || !bytes.Equal(m[2], table) {
```

Comment:
seem m here cannot be nil?

Reply:
any update? if m is nil, is it wrong?

Reply:
ping @jilieryuyi

Reply:
I think here just for validating the regex expression, `m` could be `nil` when that is mismatch, here lgtm.
cc @jilieryuyi